openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/util/ExpiringSpeedStorageSpeedCalculator.java

```diff
@@ -0,0 +1,34 @@
+package org.heigit.ors.routing.graphhopper.extensions.util;
+
+import com.graphhopper.routing.EdgeKeys;
+import com.graphhopper.routing.profiles.DecimalEncodedValue;
+import com.graphhopper.routing.util.AbstractAdjustedSpeedCalculator;
+import com.graphhopper.routing.util.FlagEncoder;
+import com.graphhopper.routing.util.SpeedCalculator;
+import com.graphhopper.util.EdgeIteratorState;
+import org.heigit.ors.routing.graphhopper.extensions.storages.ExpiringSpeedStorage;
+
+/**
+ * Simple SpeedCalculator based on a SpeedStorage.
+ * Default speed of Byte.MIN_VALUE results in original graph speed
+ */
+public class ExpiringSpeedStorageSpeedCalculator extends AbstractAdjustedSpeedCalculator {
+    private ExpiringSpeedStorage expiringSpeedStorage;
+
+    public ExpiringSpeedStorageSpeedCalculator(SpeedCalculator superSpeedCalculator, ExpiringSpeedStorage expiringSpeedStorage) {
+        super(superSpeedCalculator);
+        this.expiringSpeedStorage = expiringSpeedStorage;
+    }
+
+    public double getSpeed(EdgeIteratorState edge, boolean reverse, long time) {
+        int edgeId = EdgeKeys.getOriginalEdge(edge);
+        double modifiedSpeed = expiringSpeedStorage.getSpeed(edgeId, reverse);
+        if (modifiedSpeed == Byte.MIN_VALUE)
+            return this.superSpeedCalculator.getSpeed(edge, reverse, time);
+        return modifiedSpeed;
+    }
+
+    public void setSpeedStorage(ExpiringSpeedStorage expiringSpeedStorage) {
```

Comment:
What's the use-case of this method if `ExpiringSpeedStorage` instance is passed in the constructor?

Reply:
I think it should be possible to switch to a different storage if the need arises? Maybe a different vehicle or something? Want me to remove it?

Reply:
AFAIK typically separate instances of these classes are created for individual requests, so I don't thin there is a real use case. And if the need arises, it's trivial to add, so probably just drop it for now?

Reply:
done

Reply:
Thanks!